docs/source/tutorials/memory_optimizations.rst

```diff
@@ -0,0 +1,301 @@
+:ref:`gradient accumulation <glossary_grad_accm>`.
+
+.. _glossary_peft:
+
+Parameter Efficient Fine-Tuning (PEFT)
+--------------------------------------
+
+.. _glossary_lora:
+
+Low Rank Adaptation (LoRA)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+*What's going on here?*
+
+You can read our tutorial on :ref:`finetuning Llama2 with LoRA<lora_finetune_label>` to understand how LoRA works, and how to use it.
+Simply stated, LoRA greatly reduces the number of trainable parameters, thus saving significant gradient and optimizer
+memory during training.
+
+*Sounds great! How do I use it?*
+
+You can finetune using any of our recipes with the ``lora_`` prefix, e.g. :ref:`lora_finetune_single_device<lora_finetune_recipe_label>`. These recipes utilize
+LoRA-enabled model builders, which we support for all our models, and also use the ``lora_`` prefix, e.g.
+the :func:`torchtune.models.llama3.llama3` model has a corresponding :func:`torchtune.models.llama3.lora_llama3`.
+We aim to provide a comprehensive set of configurations to allow you to get started with training with LoRA quickly,
+just specify any config with ``_lora`` in its name, e.g:
+
+.. code-block:: bash
+
+  tune run lora_finetune_single_device --config llama3/8B_lora_single_device
+
+
+There are two sets of parameters to customize LoRA to suit your needs. Firstly, the parameters which control
+which linear layers LoRA should be applied to in the model:
+
+* ``lora_attn_modules: List[str]`` accepts a list of strings specifying which layers of the model to apply
+  LoRA to:
+
+  * ``q_proj`` applies LoRA to the query projection layer.
+  * ``k_proj`` applies LoRA to the key projection layer.
+  * ``v_proj`` applies LoRA to the value projection layer.
+  * ``output_proj`` applies LoRA to the attention output projection layer.
+
+* ``apply_lora_to_mlp: Bool`` applies LoRA to the MLP in each transformer layer.
+* ``apply_lora_to_output: Bool`` applies LoRA to the model's final output projection.
+  This is usually a projection to vocabulary space (e.g. in language models), but
+  other modelling tasks may have different projections - classifier models will project
+  to the number of classes, for example
+
+.. note::
+
+  Models which use tied embeddings (such as Gemma and Qwen2 1.5B and 0.5B) for the
+  final output projection do not support ``apply_lora_to_output``.
+
+These are all specified under the ``model`` flag or config entry, i.e:
+
+.. code-block:: bash
+
+  tune run lora_finetune_single_device --config 8B_lora_single_device  \
+  model.apply_lora_to_mlp=True \
+  model.lora_attn_modules=["q_proj", "k_proj", "v_proj"]
+
+.. code-block:: yaml
+
+  model:
+    apply_lora_to_mlp: True
+    model.lora_attn_modules: ["q_proj", "k_proj", "v_proj"]
+
+Secondly, parameters which control the scale of the impact of LoRA on the model:
+
+* ``lora_rank: int`` affects the scale of the LoRA decomposition, where ``lora_rank << in_dim`` and ``lora_rank << out_dim``
```

Comment:
prob its worth mentioning common values, eg that the bare minimum is 8, and people usually dont use more than X (not sure what X is, 128?)